### PR TITLE
Emit known classes from ExpandProto, ExpandRepeatedProto, and ExtractProto

### PR DESCRIPTION
**Pull request by bryanduxbury**

### CODE REVIEW

File: src/main/java/com/squareup/cascading2/function/AbstractExpandProto.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      Message.Builder builder = Util.builderFromMessageClass(messageClassName);
+
+      List <Descriptors.FieldDescriptor> fieldDescriptors = new ArrayList<Descriptors.FieldDescriptor>();
+      for (int i = 0; i < fieldsToExtract.length; i++) {
+        fieldDescriptors.add(builder.getDescriptorForType().findFieldByName(fieldsToExtract[i]));
+      }
+
+      fieldDescriptorsToExtract = fieldDescriptors.toArray(new Descriptors.FieldDescriptor[fieldDescriptors.size()]);
+    }
+    return fieldDescriptorsToExtract;
```
Comment (alphastorm) [inline src/main/java/com/squareup/cascading2/function/AbstractExpandProto.java:60]: nit: newline, or switch line 72 and ExtractProto.java:181 for consistency

File: src/main/java/com/squareup/cascading2/function/ExpandProto.java [hunk 2]
```diff
@@ -87,8 +68,6 @@ public ExpandProto(Class<T> messageClass, Fields fieldDeclaration, String... fie
       }
     }
```
Comment (alphastorm) [inline src/main/java/com/squareup/cascading2/function/ExpandProto.java]: nit: extra newline

**Comment by alphastorm** [comment]
LGTM after newline nits, but I'd get Nicolas or someone who reviewed the other PR to take a look since I don't have that context.

**Comment by nt** [comment]
LGTM, this was necessary. Would you mind squashing everything to one commit before merge ?

**Pull request merged**